spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/event/contractdefinition/ContractDefinitionCreated.java

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.spi.event.contractdefinition;
+
+import org.eclipse.dataspaceconnector.spi.event.Event;
+import org.eclipse.dataspaceconnector.spi.event.EventPayload;
+
+import java.util.Objects;
+
+/**
+ * Describe a new ContractDefinition creation, after this has emitted, a ContractDefinition with a certain id will be available.
+ */
+public class ContractDefinitionCreated extends Event<ContractDefinitionCreated.Payload> {
+
+    private ContractDefinitionCreated() {
+    }
+
+    public static class Builder extends Event.Builder<ContractDefinitionCreated, Payload> {
+
+        public static Builder newInstance() {
+            return new Builder();
+        }
+
+        private Builder() {
+            super(new ContractDefinitionCreated(), new Payload());
+        }
+
+        public Builder contractDefinitionId(String contractDefinitionId) {
+            event.payload.contractDefinitionId = contractDefinitionId;
+            return this;
+        }
+
+        @Override
+        protected void validate() {
+            Objects.requireNonNull(event.payload.contractDefinitionId);
+        }
+    }
+
+    public static class Payload extends EventPayload {
```

Comment:
Not directly related to this, but I just noticed that `Event.payload` is an unbounded type `P` but the `Event.Builder` is indeed bounded to `P extends EventPayload`, so wouldn't it make sense then to have `Event.payload` bounded as well?

```java
public abstract class Event<P extends EventPayload> {
    ...
}
```

Reply:
good catch, this was an oversight.